src/System.Management.Automation/engine/PSConfiguration.cs

```diff
@@ -50,6 +54,65 @@ private PowerShellConfig()
             appDataConfigDirectory = Utils.GetUserConfigurationDirectory();
         }
 
+        private string GetConfigFilePath(ConfigScope scope)
+        {
+            string result = string.Empty;
+
+            if (scope == ConfigScope.CurrentUser)
+            {
+                result = Path.Combine(appDataConfigDirectory, configFileName);
+            }
+            else
+            {
+                if (!string.IsNullOrEmpty(psCustomHomeConfigFilePath))
+                {
+                    // system wide configuration file overridden at the pwsh command-line.
+                    result = psCustomHomeConfigFilePath;
+                }
+                else
+                {
+                    result = Path.Combine(psHomeConfigDirectory, configFileName);
+                }
+            }
+            return result;
+        }
+
+        /// <summary>
+        /// Gets or sets the system wide configuration file path
+        /// </summary>
+        /// <returns>A fully qualified path to the system wide configuration file.</returns>
+        internal string SystemConfigFilePath
+        {
+            get
+            {
+                fileLock.EnterReadLock();
+                try
+                {
+                    return GetConfigFilePath(ConfigScope.SystemWide);
+                }
+                finally
+                {
+                    fileLock.ExitReadLock();
+                }
+            }
+            set
+            {
+                if (!string.IsNullOrEmpty(value) && !File.Exists(value))
+                {
+                    throw new FileNotFoundException(value);
+                }
+                fileLock.EnterWriteLock();
+                try
+                {
+                    psCustomHomeConfigFilePath = value;
+                }
+                finally
+                {
+                    fileLock.ExitWriteLock();
+                }
+            }
+        }
```

Comment:
I don't think the lock is necessary. This property should only be used during command parsing when starting the host. No Ruspace is created at that point and thus there won't be multiple threads access this property.

IMHO, this property should be setter only, and maybe we should make it only able to be set once.

Reply:
I agree with just having a setter - will change it to a method.
The locks are used around writes to the configuration files. For correctness, they should be around the access to the file paths as well.

Reply:
Removed getter and changed to Set method